src/Adapter/Feature/Repository/FeatureValueRepository.php

```diff
@@ -205,26 +212,27 @@ public function getFeatureValuesCount(?array $filters = []): int
     }
 
     /**
-     * @param int $featureValueId
+     * @param array $featureValuesIds
+     * @param int|null $langId
      *
      * @return array
      */
-    private function getFeatureValueLocalizedValues(int $featureValueId): array
+    private function getFeatureValueLocalizedValues(array $featureValuesIds, ?int $langId): array
     {
         $qb = $this->connection->createQueryBuilder();
         $qb->from($this->dbPrefix . 'feature_value_lang', 'fvl')
             ->select('fvl.*')
-            ->where('fvl.id_feature_value= :featureValueId')
-            ->setParameter('featureValueId', $featureValueId)
+            ->where('fvl.id_feature_value IN(:featureValueIds)')
```

Comment:
This is not very good, merchant will have 1000 feature values and we will be passing the whole list in the query?

The whole list can be fetched by one fast query like I did here https://github.com/PrestaShop/PrestaShop/issues/34455#issuecomment-1792211164 👍

Reply:
I think the query has been split because the initial query handles pagination and adding multiple joins in the query would completely mess with the pagination count and offsets

Reply:
Can we just make a separate method to fetch it in a fast way? Do we need all of this code to run this query? :-)

```
SELECT fvl.* FROM ps_feature_value_lang fvl
LEFT JOIN ps_feature_value fv ON fv.id_feature_value = fvl.id_feature_value
WHERE fv.id_feature = X AND fvl.id_lang = X
ORDER BY fvl.value;
```